react-native-community/react-native-webview
# Fixes Issue #1140

Fixes https://github.com/react-native-community/react-native-webview/issues/1140
Based on a solution found at: https://stackoverflow.com/questions/56460362/how-to-force-wkwebview-to-ignore-hardware-silent-switch-on-ios
I changed the code found in the linked source from Swift to Objective-C, as required by this project. WARNING: I haven't used Swift before and very limited experience with Objective-C.

- For me this seems to work, but it is not the cleanest solution in my opinion.
- It might also be possible to play generated sound (i.e. using oscillator) instead of hardcoding the silent base64 mp3 data.
- Maybe ignoring silence switch should only be done if a parameter is supplied



# Summary
Ignores the silence switch when playing audio from a `WebView`. Closes issue #1140 


## Test Plan
1. Initialise a `react-native` project, add `react-native-webview` as dependency.
2. In `App.js` add
```
<WebView injectedJavaScript={`try{var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
  var oscillator = audioCtx.createOscillator();
  oscillator.type = 'sine';
  oscillator.frequency.setValueAtTime(420, audioCtx.currentTime);
  oscillator.start();
  var oscillator_volume = audioCtx.createGain();
  oscillator_volume.gain.linearRampToValueAtTime(1, 0);
  oscillator.connect(oscillator_volume);
  oscillator_volume.connect(audioCtx.destination);
  }catch(e){}
  true;`} />
```
3. Launch the App
4. Toggle the silent switch on the device or simulator
5. Notice that when using this PR the silence switch is ignored, i.e. the sound from the oscillator (or any other webaudio) will be audible even when the silent switch is set to _on_ (silence) position
6. (optional) Do the same 5 steps, but using the code from the master branch instead and notice that when setting the silence switch to _on_ all webaudio will be muted


### What's required for testing (prerequisites)?

### What are the steps to reproduce (after prerequisites)?

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    NA     |

## Checklist




I haven't added anything to README.md, because it is just a bugfix.

I haven't added the change to CHANGELOG.md, but suppose a line could be added warning users of the new default behaviour, i.e. silence switch is ignored.
